sympy/core/numbers.py

```diff
@@ -1165,6 +1183,153 @@ def __format__(self, format_spec):
 RealNumber = Float
 
 
+class ComplexFloat(Number):
+    """Represent a complex floating-point number of arbitrary precision.
+    """
+    __slots__ = ['real', 'imag']
+
+    # each Float represents many real numbers,
+    # both erational and irrational.
+    is_rational = None
+    is_irrational = None
+    is_number = True
+
+    is_complex = True
+
+    def __new__(cls, real, imag=None, prec=None):
+        # TODO: do something with prec
+        if imag is None:
+            if isinstance(real, (complex, mpmath.mpc)):
+                real, imag = real.real, real.imag
+            else:
+                real, imag = real.as_real_imag()
+        obj = Expr.__new__(cls)
+        obj.real = Float(real, prec)
+        obj.imag = Float(imag, prec)
+        return obj
+
+    @property
+    def _prec(self):
+        return min(self.real._prec, self.imag._prec)
+
+    def _hashable_content(self):
+        return (self.real, self.imag)
+
+    def __hash__(self):
+        return super(ComplexFloat, self).__hash__()
+
+    @property
+    def is_real(self):
+        # TODO: see is_rational above
+        #return False if self.imag.is_nonzero else None
+        return self.imag.is_zero
+
+    @property
+    def is_imaginary(self):
```

Comment:
Grep for `_eval_is_`. The methods define the assumptions (instead of using `@property`).